tool/tsh/common/app_azure.go

```diff
@@ -143,7 +136,19 @@ func (a *azureApp) StartLocalProxies(ctx context.Context) error {
 	}
 
 	// HTTPS proxy mode
-	err := a.StartLocalProxyWithForwarder(ctx, alpnproxy.MatchAzureRequests, alpnproxy.WithHTTPMiddleware(azureMiddleware))
+	err := a.StartLocalProxyWithForwarder(ctx,
+		alpnproxy.MatchAzureRequests,
+		alpnproxy.WithHTTPMiddleware(azureMiddleware),
+		alpnproxy.WithOnSetCert(func(cert tls.Certificate) {
+			// Note that the PrivateKey is most likely set by api/utils/keys.TLSCertificateForSigner.
+			signer, ok := cert.PrivateKey.(crypto.Signer)
+			if ok {
+				azureMiddleware.SetPrivateKey(signer)
+			} else {
+				log.Warn("Provided tls.Certificate has no valid private key.")
```

Comment:
I think this is a critical error, right? Should we do more than just a warning?

Reply:
Ya. though, it's hard to inject a failure here. However, when the private key is not set, you get
```
Failed to connect to MSI. Please make sure MSI is configured correctly.
Get Token request returned: <Response [400]>
ERROR: exit status 1

ERROR: failed to automatically login with `az login` using identity "/subscriptions/060a97ea-3a57-4218-9be5-dba3f19ff2b5/resourcegroups/STeve-teleport/providers/Microsoft.ManagedIdentity/userAssignedIdentities/steve-azure-cli"; run with --debug for details
	exit status 1
```
where it suggests using `--debug` flag.

And with the debug flag, you would also get this warning from the azure middleware when private key is not set:
> 2024-11-08T11:22:22-05:00 WARN [AZURE_MSI] Bad MSI request: public or private key is required alpnproxy/azure_msi_middleware.go:89

I will leave it as it is.